### PR TITLE
Pipes Puzzle Implementation

### PR DESCRIPTION
**Pull request by admoore0**
I sure wish we had an italian plumber from brooklyn with a red hat and sentient dinosaur we could call on to fix these pipes!

### CODE REVIEW

File: src/puzzles/pipes/pipe.gd [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	rotation_changed.emit()
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta: float) -> void:
+	pass
```
Comment (kdreese) [inline src/puzzles/pipes/pipe.gd]:
```suggestion
```

File: src/puzzles/pipes/pipe.gd [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	for direction in directions:
+		# There's gotta be a better way
+		output.append(Vector2i(Vector2(direction).rotated(rotation)))
+
+	return output
+
+
+func on_button_pressed():
+	print("Button pressed")
```
Comment (kdreese) [inline src/puzzles/pipes/pipe.gd]: Remove prints

File: src/puzzles/pipes/pipes_puzzle.gd [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta: float) -> void:
+	pass
+
+
```
Comment (kdreese) [inline src/puzzles/pipes/pipes_puzzle.gd]:
```suggestion
```

Comment (kdreese) [reply src/puzzles/pipes/pipes_puzzle.gd]: Delet this

File: src/puzzles/pipes/pipe.gd (hunk 1 above)
Comment (thearst3rd) [inline src/puzzles/pipes/pipe.gd]: delet this :gun:

File: src/puzzles/pipes/pipes_puzzle.gd [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta: float) -> void:
+	pass
```
Comment (thearst3rd) [inline src/puzzles/pipes/pipes_puzzle.gd]: :gun:

**Review by kdreese — APPROVED** [review]

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**